### DISCUSSION TITLE
Setting domains in NextJS config

### DISCUSSION THREAD

**Discussion by herol3oy** [vercel/next.js#22350] [Help] [answered]
I use Firebase Auth in my app. When I login I need to get user's photoUrl  so at that point NextJS complains to add the URL to `next.config.js` as following:

```module.exports = {
    images: {
      domains: [
        'lh3.googleusercontent.com',
      ],
    },
  }
```
but then another user logs in and the `photoUrl`  comes from `lh4.googleusercontent.com`! What can I do in order to get all those google profile pictures from different addresses since we don't know them?

**Comment by mephysto** [comment]
I'm having similar problems with Instagram. The error I'm getting back is telling me which subdomain to add to the config.
Is it worthwhile to having a few image serving services whitelisted? Like Instagram or Facebook

**Comment by DiegoGonzalezCruz** [comment]
Hi, facing the same issue. Any help?

**Comment by leonardoTavaresM** [comment]
Hi, I'm in the same problem

**Comment by DoctorPok42** [comment]
> I use Firebase Auth in my app. When I login I need to get user's photoUrl so at that point NextJS complains to add the URL to `next.config.js` as following:
> 
> ```
>     images: {
>       domains: [
>         'lh3.googleusercontent.com',
>       ],
>     },
>   }
> ```
> 
> but then another user logs in and the `photoUrl` comes from `lh4.googleusercontent.com`! What can I do in order to get all those google profile pictures from different addresses since we don't know them?

Hello, maybe you can try to add just the domain `googleusercontent.com`
or you can see here https://nextjs.org/docs/api-reference/next/image#loader maybe it resolve your problem

**Comment by filip26** [comment, marked as answer]
With NextJS >= `v12.3` you can do this:
```javascript
  images: {
    remotePatterns: [
      {
        hostname: "*.googleusercontent.com",
      },
    ],
```
More here: https://nextjs.org/docs/messages/next-image-unconfigured-host#possible-ways-to-fix-it

**Comment by gertj1990** [comment]
I am getting hijacked, remote taken over and all my devices infected..and every time I Google scripts or look at cookies, my root folder with virtualize Linux blob, I always get pushed to here, people sharing those code. Good job guys

**Reply by Dosbodoke** [reply]
Can you explain more about this? If this is true it's a huge problem for the community because all the answers that I found has the same solution.
